src/CommunityToolkit.Maui/Alerts/Snackbar/Snackbar.windows.cs

```diff
@@ -51,27 +52,27 @@ public partial class Snackbar
 		var xmlDocument = new XmlDocument();
 		xmlDocument.LoadXml(toastContent.GetContent());
 
-		NativeSnackbar = new ToastNotification(xmlDocument);
-		NativeSnackbar.Activated += OnActivated;
-		NativeSnackbar.Dismissed += OnDismissed;
-		NativeSnackbar.ExpirationTime = DateTimeOffset.Now.Add(Duration);
+		PlatformSnackbar = new ToastNotification(xmlDocument);
+		PlatformSnackbar.Activated += OnActivated;
+		PlatformSnackbar.Dismissed += OnDismissed;
+		PlatformSnackbar.ExpirationTime = DateTimeOffset.Now.Add(Duration);
 
-		ToastNotificationManager.CreateToastNotifier().Show(nativeSnackbar);
+		ToastNotificationManager.CreateToastNotifier().Show(PlatformSnackbar);
 
 		OnShown();
 	}
 
 	void OnActivated(ToastNotification sender, object args)
 	{
-		if (NativeSnackbar is not null && Action is not null)
+		if (PlatformSnackbar is not null && Action is not null)
 		{
-			MainThread.BeginInvokeOnMainThread(Action);
+			Dispatcher.GetForCurrentThread().DispatchIfRequired(Action);
```

Comment:
While ago we discussed forcing the code to execute on `MainThread` inside the lib, and how that isn't a good practice. One reason is that this can cause some bugs and then could be hard to identify. For example, here, the `Action` will not be executed right after you step from this method, it will be queued to be executed when the `MainThread` is free, so at the time that this `Action` execute the state of the application can be invalid and this can lead to some crash or odd behavior. 

From the usage of this, this `delegate handler` will be subscribed in the thread that calls `ShowPlatform`, and the **user** must call it from the `MainThread`. What we can do here to protect the user is adding a check to make sure that we're on the `MainThread` and throw a good exception message. More than that I would say that we're shooting ourselves in the foot.

Reply:
It doesn't work when you use MainThread class: ![image](https://user-images.githubusercontent.com/33021114/160249047-f0ffd8a6-ed59-43b7-b008-ecd3a92dee90.png)

Reply:
What I mentioned is that we shouldn't use either `Dispatcher` or `MainThread`.
we should just do `Action();` or `Action.Invoke()`